### ISSUE TITLE
Probably should be ">=3.6.6, <3.12"

### ISSUE THREAD

**Issue by itsamejoshab** [cyrilou242/mockseries#47]
When attempting to install in a python 3.12 environment, we run into issues due to:

[PEP 632](https://peps.python.org/pep-0632/): Remove the distutils package. See [the migration guide](https://peps.python.org/pep-0632/#migration-advice) for advice replacing the APIs it provided. The third-party [Setuptools](https://setuptools.pypa.io/en/latest/deprecated/distutils-legacy.html) package continues to provide distutils, if you still require it in Python 3.12 and beyond.

I attempted to rectify the issue with `pip install setuptools`, however that begins to yield another thorny issue with 

`AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?`

I have ensured the most recent version of `setuptools` and `pip` but still have been unable to rectify it. So, it appears that for now, `mockseries` is limited to <3.12 versions of python until the depreciated packages can be avoided.

**Comment by cyrilou242** [comment]
Hey thanks for raising this: 

The issue is with the numpy version: 
```
Collecting mockseries
  Obtaining dependency information for mockseries from https://files.pythonhosted.org/packages/db/da/2b3387f09526cc6bd4cf9e738af0100c5374c98525ce8c5c318012aca5ed/mockseries-0.2.0-py3-none-any.whl.metadata
  Downloading mockseries-0.2.0-py3-none-any.whl.metadata (3.2 kB)
Collecting matplotlib==3.7.1 (from mockseries)
  Downloading matplotlib-3.7.1.tar.gz (38.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 38.0/38.0 MB 9.9 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting numpy==1.24.4 (from mockseries)
  Downloading numpy-1.24.4.tar.gz (10.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.9/10.9 MB 27.7 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [33 lines of output]
      Traceback (most recent call last):
        File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 112, in get_requires_for_build_wheel
          backend = _build_backend()
                    ^^^^^^^^^^^^^^^^
        File "/usr/local/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 77, in _build_backend
          obj = import_module(mod_path)
                ^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 994, in exec_module
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "/tmp/pip-build-env-mwmqzvha/overlay/lib/python3.12/site-packages/setuptools/__init__.py", line 16, in <module>
          import setuptools.version
        File "/tmp/pip-build-env-mwmqzvha/overlay/lib/python3.12/site-packages/setuptools/version.py", line 1, in <module>
          import pkg_resources
        File "/tmp/pip-build-env-mwmqzvha/overlay/lib/python3.12/site-packages/pkg_resources/__init__.py", line 2172, in <module>
          register_finder(pkgutil.ImpImporter, find_on_path)
                          ^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

[notice] A new release of pip is available: 23.2.1 -> 23.3.2
[notice] To update, run: pip install --upgrade pip
```


NumPy 1.26.0 is compatible with python `3.9-3.12`.
NumPy 1.25.0 is compatible with python `3.9-3.11.`.
NumPy 1.24.0 is compatible with python `3.8-3.11`.
NumPy 1.19.0 is compatible with python `3.6-??`.

Mockseries is actually now using Numpy 1.24 so it's already >=3.8 only.

We could upgrade Numpy to 1.26 and maintain a compatibility matrix in the doc. 
Something like: 
| Mockseries version | Python versions |
| --- | --- |
| 0.3.0 | 3.9 - 3.12|
| 0.2.0 | 3.8 - 3.11|
| 0.1.4 | 3.6 - 3.8|

What do you think @itsamejoshab ?

**Comment by cyrilou242** [comment]
closing. `0.3.0` is released with support for python 12
